lib/matplotlib/backends/backend_pdf.py

```diff
@@ -437,7 +437,7 @@ def __init__(self, filename, metadata=None):
         Parameters
         ----------
 
-        filename : file-like object or string
+        filename : str or path-like or file-like
```

Comment:
Is chaining `or` like this the style we want?   Could do as `str, path-like, or file-like` which reads better to me....

Reply:
That's how the [numpydoc docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html) describes it:

> shape : int or tuple of int

Currently we use a comma for simple tuples, e.g. `center  : float, float`.

I don't have a strong opinion on this, but if we change that, we need another convention for tuple to avoid ambiguity (could be `(float, float)`). Everything should be changed consistently throughout the docs and we should document it in "writing documentation".

Reply:
If you use the Oxford comma it’s not ambiguous: 

A, B, or c

A, B or c

Though I would tend to enclose tuples in brackets just to be clear